src/RadixDlt.NetworkGateway.PostgresIntegration/Metrics/SqlQueryMetricsHelper.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+namespace RadixDlt.NetworkGateway.PostgresIntegration.Metrics;
+
+public static class SqlQueryMetricsHelper
+{
+    public const string QueryNameTag = "QueryName";
+
+    public static string GetQueryNameValue(
+        string operationName,
+        string methodName)
+    {
+        if (!operationName.All(char.IsLetterOrDigit))
+        {
+            throw new ArgumentException($"{nameof(operationName)} is expected to be alphanumeric. Got: {operationName}");
+        }
+
+        if (!methodName.All(char.IsLetterOrDigit))
```

Comment:
This one is dangerous as method names can contain `_` and while not supper common in C# it's still a valid one.,